### PR TITLE
KBV-342: Address CRI audit events to send to TxMA

### PR DESCRIPTION
**Pull request by techaman-m**
Implement event payloads that the Address CRI sends to TxMA at 3 different instances during the address-cri flow:

- before a session is written to the Session table - IPV_ADDRESS_CRI_START
- When an address is added in the PUT /address API call - IPV_ADDRESS_CRI_REQUEST_SENT
- When the final VC is created in the issue credential lambda - IPV_ADDRESS_CRI_VC_ISSUED

### What changed

For slice 2 the above events will be like:

```
{
  timestamp: 1609462861,
  event_name: "IPV_ADDRESS_CRI_START"
}
```

### Issue tracking

- [KBV-342](https://govukverify.atlassian.net/browse/KBV-342)

Messages in Queue:
image

Message when CRI request sent:
image

### CODE REVIEW

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java [hunk 1]
```diff
@@ -20,28 +19,24 @@ public AuditService(AmazonSQS sqs, ConfigurationService configurationService) {
...
             throw new SqsException(e);
         }
     }
 
-    private String generateMessageBody(AuditEventTypes eventType, UUID sessionId, String clientId)
-            throws JsonProcessingException {
+    private String generateMessageBody(AuditEventTypes eventType) throws JsonProcessingException {
         AuditEvent auditEvent = new AuditEvent();
         Instant now = Instant.now();
         int instant = (int) now.getEpochSecond();
```
[the remark this answers is not in the document]
Comment (techaman-m) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java]: @KaraokeStu , nice one.

I have changed as per comments

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java [hunk 2]
```diff
@@ -20,30 +21,26 @@ public AuditService(AmazonSQS sqs, ConfigurationService configurationService) {
...
     }
 
-    private String generateMessageBody(AuditEventTypes eventType, UUID sessionId, String clientId)
-            throws JsonProcessingException {
+    private String generateMessageBody(AuditEventTypes eventType) throws JsonProcessingException {
         AuditEvent auditEvent = new AuditEvent();
-        Instant now = Instant.now();
-        int instant = (int) now.getEpochSecond();
-        auditEvent.setTimestamp(instant);
+        auditEvent.setTimestamp(new Date());
```
Comment (idrop) [inline lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java:38]: could we drop the timestamp setter and just instantiate a new Date in the AuditEvent default constructor? I can't think of reason we wouldn't want to set the timestamp to anything other than now.

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java [hunk 3]
```diff
@@ -20,30 +21,26 @@ public AuditService(AmazonSQS sqs, ConfigurationService configurationService) {
...
         auditEvent.setEvent(eventType);
-        auditEvent.setClientId(clientId);
-        auditEvent.setEventId(sessionId.toString());
-        auditEvent.setTimestampFormatted(now.toString());
-        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectMapper objectMapper =
+                new ObjectMapper()
+                        .registerModule(new Jdk8Module())
+                        .registerModule(new JavaTimeModule());
```
[the remark this answers is not in the document]
Comment (techaman-m) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java]: `@InjectMocks` seems to work when the constructor has simple mocks which can be straightforward to use. Tried to use it here but the internal state of capture was not being set.

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java (hunk 2 above)
Comment (techaman-m) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AuditService.java]: Added to constructor

**Review by idrop — CHANGES_REQUESTED** [review]
I'm creating a new PR for this tha will work post-git-submodules

**Pull request closed without merging**